### PR TITLE
[quant][graphmode] Add warning for debug option for add_scalar/mul_scalar

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40414 [quant][graphmode] Enable inplace option for top level API
* #40384 [quant][graphmode][refactor] InsertQuantDeQuantHelper
* **#40383 [quant][graphmode] Add warning for debug option for add_scalar/mul_scalar**

Summary:
debug option is not supported for these cases, so we print a warning if it occurs

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22164071](https://our.internmc.facebook.com/intern/diff/D22164071)

### CODE REVIEW

**Review by vkuzo — APPROVED** [review]
> we don't check the numerical consistency for add_scalar since it's not supported

just curious, what's the context on why it's not supported?

**Comment by jerryzh168** [comment]
> > we don't check the numerical consistency for add_scalar since it's not supported
> 
> just curious, what's the context on why it's not supported?

oh, the reason is written in the comments

File: torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp [hunk 1]
```diff
@@ -1024,6 +1024,16 @@ void InsertQuantDeQuantHelper::propagateQuantizationOps(Block* block) {
           propagateQParams(output, *inputs, /* is_scalar */ false, qparams_opt);
         }
       }
+    } else if (isPropagateQuantNode(n)) {
+      // Print warning for add_scalar/mul_scalar when debug is enabled
```
Comment (jerryzh168) [inline torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp:1028]: @vkuzo

**Review by raghuramank100 — APPROVED** [review]

**Comment by vkuzo** [comment]
> > > we don't check the numerical consistency for add_scalar since it's not supported
> > 
> > 
> > just curious, what's the context on why it's not supported?
> 
> oh, the reason is written in the comments

thanks, wasn't sure if it was that comment or something else, got it

**Pull request closed without merging**